cub/device/dispatch/dispatch_scan.cuh

```diff
@@ -387,12 +389,12 @@ struct DispatchScan:
     CUB_RUNTIME_FUNCTION __host__  __forceinline__
     cudaError_t Invoke()
```

Comment:
Host-side `Invoke` function template is getting instantiated for _all_ tuning policies in the chain. Hence, the explicit kernel template instantiation for `DeviceScanKernel<Policy ...` below would be instantiated for all `Policy`.
 
Instead, the idea is to instantiate the `DeviceScanKernel` kernel template always with the top of the tuning policy chain and then, in the device code, identify and apply the tuning policy for the GPU architecture of the current compilation pass.